### DISCUSSION TITLE
Passing data from client component to server component.

### DISCUSSION THREAD

**Discussion by bonraton** [vercel/next.js#57427] [App Router]
Hello everyone.
i have this router structure with Next 13:

- /app
- - /[lang]
- - -/events
- - - - [eventId]
- - - - - - page.tsx

In page.tsx I fetch event and render component

`export default async function Page({ params }: { params: WithLang<{ eventId: string }> }) {`
   ` const pEvent = (await events$get$details(params.eventId))!;`

`// Other code`

Now I am trying to avoid ID in URL string to make it readabale.

from 
`http://localhost:3000/en/events/2c2b8eba-953f-4ead-9e53-7ae7f885aafd`

to
`http://localhost:3000/en/events/awesome_event_name`

But I still need an ID to fetch data.
I tried to pass the ID with a search params, but it also must be in url to get it.

Do anyone know how to solve it?

**Comment by icyJoseph** [comment]
In general that pattern is not supported, just yet. You probably want to use query params, or maybe a cookie? and call router.refresh() when you set the cookie?

**Reply by bonraton** [reply]
I didn't try that one, thanks for advise. Put event Id to cookie sounds not the best. But it would work =)

UPD: Cookies solved my problem

**Comment by landsman** [comment]
why do you even need a client for that? just change `eventId` to `slug` and operate with this path as a key which you want to find in your database.
